math/n_bonacci.cpp

```diff
@@ -0,0 +1,107 @@
+/**
+ * @file
+ * @brief Implementation of [N-bonacci](http://oeis.org/wiki/N-bonacci_numbers)
+ * Series
+ *
+ * @details
+ * In general, in N-bonacci sequence,
+ * we generate sum of preceding N numbers from the next term.
+ *
+ * For example, a 3-bonacci sequence is the following:
+ * 0, 0, 1, 1, 2, 4, 7, 13, 24, 44, 81
+ * In this code we take N and M as input where M is the number of terms
+ * to be printed of the N-bonacci series
+ *
+ * @author [Swastika Gupta](https://github.com/swastyy)
+ */
+#include <algorithm>  /// for std::is_sorted, std::swap
+#include <cassert>    /// for assert
+#include <iostream>   /// for io operations
+#include <vector>     /// for std::vector
+
+namespace math {
+/**
+ * @namespace math
+ * @brief Function for n-bonacci series
+ */
+namespace n_bonacci {
+/**
+ * @brief The main function implements n-bonacci series
+ * @tparam T type of array
+ * @param n,m m is the number of terms in n-bonacci sequence
+ * @returns the n-bonacci sequence as vector array
+ */
+template <typename T>
+std::vector<T> N_bonacci(const std::vector<T> &in_arr) {
+    /**
+     * This function finds N-bonacci series for
+     * @param n value and @param m terms
```

Comment:
That is not a parameter. Also, please use one-line descriptions **only** inside functions.

Reply:
Done - both the description and the github actions @Panquesito7